vars/runPerfTestScript.groovy

```diff
@@ -0,0 +1,47 @@
+void call(Map args = [:]) {
+    String jobName = args.jobName ?: 'distribution-build-opensearch'
+    lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.bundleManifest))
+    String artifactRootUrl = buildManifest.getArtifactRootUrl(jobName, args.buildId)
+
+    install_npm()
+    install_dependencies()
+    install_opensearch_infra_dependencies()
+    withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
+        s3Download(file: "config.yml", bucket: "${ARTIFACT_BUCKET_NAME}", path: "${PERF_TEST_CONFIG_LOCATION}/config.yml", force: true)
+    }
+
+    sh([
+        './test.sh',
+        'perf-test',
+        args.security ? "--stack test-single-security-${args.buildId}" :
+        "--stack test-single-${args.buildId}",
+        "--bundle-manifest ${args.bundleManifest}",
+        "--config config.yml",
+        args.security ? "--security" : ""
+    ].join(' '))
+}
+
+void install_opensearch_infra_dependencies() {
```

Comment:
Let's use a docker image to avoid downloading all the dependencies.

Reply:
These dependencies are not a part of any docker image and I don't think it should be. They are specific to mensor lib. This is similar to pip install for all our libraries in this repo.

Reply:
I see. What about `npm` `nodejs` `aws-cdk` etc?
https://github.com/opensearch-project/opensearch-build/blob/7affa4d4a410100431344e930d57aec042ba16c6/vars/runPerfTestScript.groovy#L33-L47

Apart from the `pipenv` installs, we should have the rest in a docker image

Reply:
I have removed the nodejs install. 
`aws-cdk` and `cdk-assume-role-credential-plugin` are specific to performance tests.
I think they are appropriate being in this particular script.

Reply:
@kotwanikunal - I believe gcc too right?
@peterzhuamazon - what are your thoughts on this?

Reply:
thanks!